src/redash_stmo/health.py

```diff
@@ -0,0 +1,108 @@
+import os
+import json
+import time
+import redash
+
+from redash_stmo import settings
+
+from flask import jsonify
+from flask_login import login_required
+
+from celery.utils.log import get_task_logger
+from redash import models, redis_connection, statsd_client
+from redash.worker import celery
+from redash.utils import parse_human_time
+from redash.monitor import get_status as original_get_status
+from redash.query_runner import BaseQueryRunner
+from redash.handlers.base import routes
+from redash.permissions import require_super_admin
+
+logger = get_task_logger(__name__)
+
+
+def store_health_status(data_source_id, data_source_name, query_text, data):
+    key = "data_sources:health"
+
+    cache = json.loads(redis_connection.get(key) or '{}')
+    if data_source_id not in cache:
+        cache[data_source_id] = {
+            "metadata": {"name": data_source_name},
+            "queries": {}
+        }
+    cache[data_source_id]["queries"][query_text] = data
+
+    cache[data_source_id]["status"] = "SUCCESS"
+    for query_status in cache[data_source_id]["queries"].values():
+        if query_status["status"] == "FAIL":
+            cache[data_source_id]["status"] = "FAIL"
+            break
+
+    redis_connection.set(key, json.dumps(cache))
+
+
+@celery.task(name="redash_stmo.health.update_health_status", time_limit=90, soft_time_limit=60)
+def update_health_status():
+    for data_source in models.DataSource.query:
+        logger.info(u"task=update_health_status state=start ds_id=%s", data_source.id)
+
+        runtime = None
+        query_text = data_source.query_runner.noop_query
+        #custom_queries = settings.CUSTOM_HEALTH_QUERIES
+        ds_id = str(data_source.id)
+
+        custom_query_env_var = "REDASH_CUSTOM_HEALTH_QUERIES_{data_source_id}".format(data_source_id=ds_id)
+        custom_query = os.environ.get(custom_query_env_var, "")
+        query_text = custom_query or query_text
+
+        #if custom_queries and ds_id in custom_queries:
+        #    query_text = custom_queries[ds_id]
+
+        try:
+            start_time = time.time()
+            test_connection(data_source.query_runner, query_text)
+            runtime = time.time() - start_time
+        except NotImplementedError:
+            logger.warning(u"Unable to compute health status without test query for %s", data_source.name)
+            continue
+        except Exception as e:
+            logger.warning(u"Failed health check for the data source: %s", data_source.name, exc_info=1)
+            statsd_client.incr('update_health_status.error')
+            logger.info(u"task=update_health_status state=error ds_id=%s runtime=%.2f", data_source.id, time.time() - start_time)
+
+        status = {
+            "status": "SUCCESS" if runtime is not None else "FAIL",
```

Comment:
Let's remove the double negative and write `"FAIL" if runtime is None else "SUCCESS",`